projects/RabbitMQ.Client/client/impl/ConsumerDispatching/ConsumerDispatcher.cs

```diff
@@ -22,30 +22,20 @@ protected override async Task ProcessChannelAsync()
                 {
                     try
                     {
-                        var consumer = work.Consumer;
-                        var consumerTag = work.ConsumerTag;
-                        switch (work.WorkType)
+                        var task = work.WorkType switch
                         {
-                            case WorkType.Deliver:
-                                consumer.HandleBasicDeliver(consumerTag, work.DeliveryTag, work.Redelivered, work.Exchange, work.RoutingKey, work.BasicProperties, work.Body);
-                                break;
-                            case WorkType.Cancel:
-                                consumer.HandleBasicCancel(consumerTag);
-                                break;
-                            case WorkType.CancelOk:
-                                consumer.HandleBasicCancelOk(consumerTag);
-                                break;
-                            case WorkType.ConsumeOk:
-                                consumer.HandleBasicConsumeOk(consumerTag);
-                                break;
-                            case WorkType.Shutdown:
-                                consumer.HandleModelShutdown(_model, work.Reason);
-                                break;
-                        }
+                            WorkType.Deliver => work.Strategy.DispatchBasicDeliver(work.ConsumerTag, work.DeliveryTag, work.Redelivered, work.Exchange, work.RoutingKey, work.BasicProperties, work.Body),
```

Comment:
This will result in another interface call, which negatively impacts the performance in a perf critical path.